src/onion/response.c

```diff
@@ -615,6 +615,20 @@ void onion_response_add_cookie(onion_response *res, const char *cookiename, cons
 	if (flags&OC_SECURE)
 		pos+=snprintf(data+pos, sizeof(data)-pos, "; Secure");
 
+	switch (flags & (OC_SAMESITE_NONE | OC_SAMESITE_LAX | OC_SAMESITE_STRICT)) {
```

Comment:
Related to comment at onion_response_flags_e, I would use simple ifs as above as this switch does not really work as intended.

The simple ifs can be something like:

```c
if (flags&OC_SAMESITE_NONE) { 
... 
} else if (flags&OC_SAMESITE_LAX) {
  ...
} else if (flags&OC_SAMESITE_STRICT) {
 ...
}
```

And optionally some warning if several selected:

```c
if (check_if_several_bits_are_set...){
  WARNING("Must set only one of OC_SAMESITE_NONE, OC_SAMESITE_LAX or OC_SAMESITE_STRICT");
}
```